packages/react/src/components/Popup/Popup.tsx

```diff
@@ -251,9 +254,29 @@ export default class Popup extends AutoControlledComponent<ReactProps<PopupProps
             useCapture: true,
           },
         )
+        this.outsideKeySubscription = EventStack.subscribe(
+          'keydown',
+          e => {
+            const keyCode = keyboardKey.getCode(e)
+
+            const matchingKey = keyCode === keyboardKey.Enter || keyboardKey.Spacebar
+            const isOutsidePopupElement =
+              this.popupDomElement && !this.popupDomElement.contains(e.target)
```

Comment:
can you extract this logic from `updateOutsideClickSubscription` so we can reuse here?